maas/resource_maas_vm_host_test.go

```diff
@@ -0,0 +1,83 @@
+package maas_test
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"terraform-provider-maas/maas"
+	"terraform-provider-maas/maas/testutils"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccMAASVMHost_DeployParams(t *testing.T) {
+	// Get machine identifier from env var
+	machineIdentifier := os.Getenv("TF_ACC_VM_HOST_MACHINE")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testutils.PreCheck(t, nil) },
+		Providers:    testutils.TestAccProviders,
+		CheckDestroy: testAccCheckMAASVMHostDestroy,
+		ErrorCheck:   func(err error) error { return err },
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMaasVMHostDeployParams(machineIdentifier),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("maas_vm_host.test-vm-host", "type", "lxd"),
+					resource.TestCheckResourceAttr("maas_vm_host.test-vm-host", "machine", machineIdentifier),
+					resource.TestCheckResourceAttr("maas_vm_host.test-vm-host", "deploy_params.0.distro_series", "noble"),
+					resource.TestCheckResourceAttr("maas_vm_host.test-vm-host", "deploy_params.0.enable_hw_sync", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccMaasVMHostDeployParams(machineIdentifier string) string {
+	return fmt.Sprintf(`
+	resource "maas_vm_host" "test-vm-host" {
+	  machine = %q
+	  type    = "lxd"
+
+	  deploy_params {
+		  distro_series    = "ubuntu/noble"
+		  enable_hw_sync   = true
+		  user_data        = "#!/bin/bash\necho 'Hello from cloud-init'"
+	  }
+	}
+	`, machineIdentifier)
+}
+
+func testAccCheckMAASVMHostDestroy(s *terraform.State) error {
```

Comment:
let's also confirm that the machine is released in MAAS and back in Ready state.